### PR TITLE
#916 #668 No IntelliSense in Interactive Window with ES6 Mode

### LINKED ISSUE #916: No Intellisense in interactive Window With ES6 Intellisense
Interactive widow should show intellisense with ES6 preview Intellisense.

Here's the current behavior with Full intellisense:

image

and with ES6 Preview:

image

### PR DESCRIPTION
**Pull request by mousetraps**
* Initialize loose analyzer with Quick IntelliSense if Preview mode is on.
* Don't display errors when ES6 Mode is on
* Don't update status bar for implicit projects

close #916, address part of #668

### CODE REVIEW

File: Nodejs/Product/Nodejs/NodejsPackage.cs [hunk 1]
```diff
@@ -661,6 +661,21 @@ internal sealed partial class NodejsPackage : CommonPackage {
             LogLooseFileAnalysisLevel();
         }
 
+        private VsProjectAnalyzer CreateLooseVsProjectAnalyzer() {
```
Comment (mjbvz) [inline Nodejs/Product/Nodejs/NodejsPackage.cs]: Are there any non-repl flows through `GetAnalyzer` that could be regressed by changing `DefaultAnalyzer`?

File: Nodejs/Product/Nodejs/NodejsPackage.cs [hunk 2]
```diff
@@ -661,6 +661,21 @@ internal sealed partial class NodejsPackage : CommonPackage {
...
+            // it will throw to JSLS. This means that currently, loose files will be handled by the JSLS editor,
+            // rather than the NodeLS editor (which means that the loose project analyzer will not be invoked
+            // in these scenarios.)
+            //
+            // Because Salsa doesn't support the stitching together inputs from the surrounding text view,
+            // we do not throw to Salsa from the REPL window. However, in order to provide a workable editing
+            // experience within the REPL context, we initialize the loose analyzer with Quick IntelliSense
+            // during ES6 mode.
+            var analysisLevel = IntellisenseOptionsPage.AnalysisLevel == AnalysisLevel.Preview ? AnalysisLevel.NodeLsMedium : IntellisenseOptionsPage.AnalysisLevel;
+            return new VsProjectAnalyzer(analysisLevel, false);
```
Comment (mjbvz) [inline Nodejs/Product/Nodejs/NodejsPackage.cs]: With this change, we are taking an explicit dependency back on our old AnalysisEngine, whereas we currently are almost in a state where we could delete all that code, simplifying the maintenance and development of NTVS considerably, therefore improving product quality in the long run.

I don't have any problem with making this change for 1.2, but want to ensure we have a plan in place to fix the underlying issues that are preventing us from using a modern intellisense engine in the repl window. Is this the case?

**Comment by mjbvz** [comment]
LGTM. Merge when ready, but please respond to the more general comments/questions I had about this change and planning.

File: Nodejs/Product/Nodejs/NodejsPackage.cs (hunk 2 above)
Comment (mousetraps) [inline Nodejs/Product/Nodejs/NodejsPackage.cs]: Just added #668 to our On Radar milestone so it doesn't fall through the cracks. We won't be addressing it for Dev14, but certainly for Dev15

File: Nodejs/Product/Nodejs/NodejsPackage.cs (hunk 1 above)
Comment (mousetraps) [inline Nodejs/Product/Nodejs/NodejsPackage.cs]: The two usage scenarios are:
* opening loose files, which now just throws to JSLS
* interactive window

**Pull request merged**